### PR TITLE
fix volume & categorize endpoints

### PR DESCRIPTION
**Pull request by green-jay**

### CODE REVIEW

**Comment by green-jay** [comment]
@vgantchev any idea why hydradx-ui/v1/stats/current/price/0 returns nothing?

**Comment by vgantchev** [comment]
also /1 doesn't return anything - idk maybe no data?

File: queries/defillama/v1/tvl.sql [hunk 1]
```diff
@@ -16,7 +16,7 @@ FROM
   JOIN omnipool_asset oa ON leb.height = oa.block
   JOIN token_metadata tm ON oa.asset_id = tm.id
 WHERE CASE
-      WHEN :asset::text IS NOT NULL
```
Comment (vgantchev) [inline queries/defillama/v1/tvl.sql:19]: @green-jay is this change intentional?

File: queries/defillama/v1/volume.sql [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 SELECT 
   CASE
-      WHEN :asset::text IS NOT NULL
```
Comment (vgantchev) [inline queries/defillama/v1/volume.sql:7]: also this one?

**Review by vgantchev — COMMENTED** [review]
Did some reorganizing to make the endpoints more restful + added missing cache settings. Here's what changed:

/historical/ => /charts/
/current/price => /price (same for volume and tvl)
/stats/alltime/volume => /stats/volume/alltime/:assetId? (it is now part of the same endpoint file)

Before merging we need to check loading speeds. Also some irregularities spotted where some of the new endpoints return the following msg before cache is warm - perhaps some SQL operation taking place at first load?

`ERR value is not an integer or out of range`

**Comment by green-jay** [comment]
> also /1 doesn't return anything - idk maybe no data?

there is data for 0. 1 is currently expected to return nothing as its special case = lrna (but can be also added to this query so it is complete, currently under lrna separate endpoint)

File: queries/defillama/v1/tvl.sql (hunk 1 above)
Comment (green-jay) [reply queries/defillama/v1/tvl.sql:19]: yes but it can stay either way, this was just part of changing asset to integer everywhere more explicitely

**Comment by green-jay** [comment]
> Did some reorganizing to make the endpoints more restful + added missing cache settings. Here's what changed:
> 
> /historical/ => /charts/ /current/price => /price (same for volume and tvl) /stats/alltime/volume => /stats/volume/alltime/:assetId? (it is now part of the /volume endpoint file)
> 
> Before merging we need to check loading speeds. Also some irregularities spotted where some of the new endpoints return the following msg before cache is warm - perhaps some SQL operation taking place at first load?
> 
> `ERR value is not an integer or out of range`

i did see this error as well, started happening after changing asset from string to integer, not sure why

File: queries/defillama/v1/tvl.sql (hunk 1 above)
Comment (vgantchev) [reply queries/defillama/v1/tvl.sql:19]: are you sure? According to the defillama endpoint specification, it passes ticker

**Comment by vgantchev** [comment]
> > Did some reorganizing to make the endpoints more restful + added missing cache settings. Here's what changed:
> > /historical/ => /charts/ /current/price => /price (same for volume and tvl) /stats/alltime/volume => /stats/volume/alltime/:assetId? (it is now part of the /volume endpoint file)
> > Before merging we need to check loading speeds. Also some irregularities spotted where some of the new endpoints return the following msg before cache is warm - perhaps some SQL operation taking place at first load?
> > `ERR value is not an integer or out of range`
> 
> i did see this error as well, started happening after changing asset from string to integer, not sure why

Can it be related to the data not being ready? Does it use some PG cache under the hood?

File: queries/defillama/v1/tvl.sql (hunk 1 above)
Comment (green-jay) [reply queries/defillama/v1/tvl.sql:19]: ah ofc my bad, that should be reverted

Comment (green-jay) [reply queries/defillama/v1/tvl.sql:19]: Defillama TVL endpoint should be removed altogether, they are not using it. They are using only /volume endpoint

File: queries/defillama/v1/volume.sql (hunk 2 above)
Comment (green-jay) [reply queries/defillama/v1/volume.sql:7]: correct, this should be reverted

**Comment by green-jay** [comment]
> > > Did some reorganizing to make the endpoints more restful + added missing cache settings. Here's what changed:
> > > /historical/ => /charts/ /current/price => /price (same for volume and tvl) /stats/alltime/volume => /stats/volume/alltime/:assetId? (it is now part of the /volume endpoint file)
> > > Before merging we need to check loading speeds. Also some irregularities spotted where some of the new endpoints return the following msg before cache is warm - perhaps some SQL operation taking place at first load?
> > > `ERR value is not an integer or out of range`
> > 
> > 
> > i did see this error as well, started happening after changing asset from string to integer, not sure why
> 
> Can it be related to the data not being ready? Does it use some PG cache under the hood?

i wasn't able to replicate the error anymore, all seems to be working fine

**Review by vgantchev — APPROVED** [review]
